backend/utils/gameLogic.js

```
@@ -67,7 +87,11 @@ function tryToStealWord(word, pot, socketId, gameState) {
   // Try each existing word to see if it can be used for stealing
   for (const existingWord of allPlayerWords) {
     // Check if this is an exact anagram - if so, skip this word
-    if (isAnagram(existingWord, word) && existingWord !== word) continue;
+    if (
+      (isAnagram(existingWord, word) && existingWord !== word) ||
+      sharesSameRoot(existingWord, word)
```

We should propagate the error and let clients know they failed because it's a shared root. Feel free to leave it as a follow up issue.